File: sources/com/lumiyaviewer/lumiya/slproto/messages/CheckParcelSales.java
package com.lumiyaviewer.lumiya.slproto.messages;

import com.google.common.primitives.UnsignedBytes;
import com.lumiyaviewer.lumiya.slproto.SLMessage;
import java.nio.ByteBuffer;
import java.util.ArrayList;

public class CheckParcelSales extends SLMessage {
    public ArrayList<RegionData> RegionData_Fields = new ArrayList<>();

    public static class RegionData {
        public long RegionHandle;
    }

    public CheckParcelSales() {
        this.zeroCoded = false;
    }

    public int CalcPayloadSize() {
        return (this.RegionData_Fields.size() * 8) + 5;
    }

    public void Handle(SLMessageHandler sLMessageHandler) {
        sLMessageHandler.HandleCheckParcelSales(this);
    }

    public void PackPayload(ByteBuffer byteBuffer) {
        byteBuffer.putShort(-1);
        byteBuffer.put((byte) 0);
        byteBuffer.put((byte) -31);
        byteBuffer.put((byte) this.RegionData_Fields.size());
        for (RegionData regionData : this.RegionData_Fields) {
            packLong(byteBuffer, regionData.RegionHandle);
        }
    }

    public void UnpackPayload(ByteBuffer byteBuffer) {
        byte b = byteBuffer.get() & UnsignedBytes.MAX_VALUE;
        for (int i = 0; i < b; i++) {
            RegionData regionData = new RegionData();
            regionData.RegionHandle = unpackLong(byteBuffer);
            this.RegionData_Fields.add(regionData);
        }
    }
}

File: Linkpoint/src/main/java/com/linkpoint/slproto/messages/CheckParcelSales.kt
package com.linkpoint.slproto.messages

import com.google.common.primitives.UnsignedBytes
import com.linkpoint.slproto.SLMessage
import java.nio.ByteBuffer

class CheckParcelSales : SLMessage() {
    val RegionData_Fields = ArrayList<RegionData>()

    class RegionData {
        var RegionHandle: Long = 0
    }

    init {
        zeroCoded = false
    }

    override fun CalcPayloadSize(): Int = (RegionData_Fields.size * 8) + 5

    override fun Handle(handler: SLMessageHandler) {
        handler.HandleCheckParcelSales(this)
    }

    override fun PackPayload(buffer: ByteBuffer) {
        buffer.putShort(-1)
        buffer.put(0.toByte())
        buffer.put((-31).toByte())
        buffer.put(RegionData_Fields.size.toByte())
        for (regionData in RegionData_Fields) {
            packLong(buffer, regionData.RegionHandle)
        }
    }

    override fun UnpackPayload(buffer: ByteBuffer) {
        val count = buffer.get().toInt() and UnsignedBytes.MAX_VALUE.toInt()
        for (i in 0 until count) {
            val regionData = RegionData()
            regionData.RegionHandle = unpackLong(buffer)
            RegionData_Fields.add(regionData)
        }
    }
}